T24348

tchx84 opened the pull request:
https://phabricator.endlessm.com/T24348

tchx84 committed:
HackUnlock: remove unncessary callback

We should _never_ display the reset button
for this toolbox, therefore makes no sense
to connect to this signal.

https://phabricator.endlessm.com/T24348

ptomato commented on src/toolbox.js:
```diff
@@ -61,6 +64,10 @@ var Toolbox = GObject.registerClass({
         this.setBusy(false);
+
+        this._gameState = GameState.getDefault();
+        this._gameState.connect('g-signal', this._onStateChanged.bind(this));
```
If you use the (unfortunately JS-only) signal that the proxy object creates, then you can do the state change without another sync call to get the state.

See https://github.com/endlessm/hack-hackdex/blob/master/com.endlessm.Hackdex_chapter_one/controller/meshDev.js#L41 for an example

tchx84 replied:
I can do the same thing with the current approach :) But since I _have_ to read it while constructing the toolbox, and whole try/unpack boilerplate hurts, is just looked simpler to do it this way. But I can surely save that one call.

ptomato commented on src/toolbox.js:
```diff
@@ -113,4 +120,26 @@ var Toolbox = GObject.registerClass({
+        const key = params.get_child_value(0).deep_unpack();
+        if (key !== RESET_BUTTON_KEY)
+            return;
+        this._updateResetButton();
```
Or alternatively, get the value from `get_child_value(1)` and update the state based on that. It just seems wasteful to do another sync call if you already have the value here.

tchx84 replied:
Yeah, this is what I meant. But, I am curious about how you approach works.

ptomato reviewed (changes requested):
One comment for saving a sync call

tchx84 committed:
gameState: add getDictValueSync helper

So we don't repeat this same pattern all
over the place.

https://phabricator.endlessm.com/T24348

tchx84 committed:
toolbox: reset button visible on GSS

The default visibility of the toolbar reset button
should be not-visible, until quests enable it, via
the GSS.

https://phabricator.endlessm.com/T24348

tchx84 committed:
locksManager: re-use getDictValueSync helper

https://phabricator.endlessm.com/T24348

ptomato commented on src/locksManager.js:
```diff
@@ -47,14 +47,7 @@ var LocksManager = GObject.registerClass({
-        } catch (error) {
-            return false;
-        }
+        return !this._proxy.getDictValue(lock, 'locked', true);
```
Seems like the code you are replacing had the default value equal to `false`, now it is `true`?

tchx84 replied:
@ptomato yes, and that is correct. !true === false, and that is what should return as default value (note the negation).

ptomato commented on src/gameState.js:
```diff
@@ -52,6 +62,9 @@ var getDefault = (function () {
                     });
                 });
             };
+            Object.assign(defaultGameStateProxy, {
```
Optional; is there a reason to not define it inline like the other methods here?

tchx84 replied:
Not really, I can fix it.
